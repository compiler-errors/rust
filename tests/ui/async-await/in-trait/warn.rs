// edition: 2021

#![feature(async_fn_in_trait)]
#![deny(async_fn_in_trait)]

trait Foo {
    async fn not_send();
    //~^ ERROR
}

fn main() {}
