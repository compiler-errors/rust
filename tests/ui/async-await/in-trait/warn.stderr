error: usage of `async fn` in trait is discouraged because they do not automatically have auto trait bounds
  --> $DIR/warn.rs:7:5
   |
LL |     async fn not_send();
   |     ^^^^^
   |
   = note: you can suppress this lint if you plan to use the trait locally, for concrete types, or do not care about auto traits like `Send` on the future
note: the lint level is defined here
  --> $DIR/warn.rs:4:9
   |
LL | #![deny(async_fn_in_trait)]
   |         ^^^^^^^^^^^^^^^^^
help: you can alternatively desugar the `async fn` and any add additional traits such as `Send` to the signature
   |
LL -     async fn not_send();
LL +     fn not_send() -> impl std::future::Future<Output = ()> + Send;
   |

error: aborting due to previous error

